velox/functions/sparksql/specialforms/FromJson.cpp

```diff
@@ -267,6 +273,56 @@ struct ExtractJsonTypeImpl {
     }
   };
 
+  static simdjson::error_code castJsonToDate(
+      Input value,
+      exec::GenericWriter& writer) {
+    SIMDJSON_ASSIGN_OR_RAISE(auto type, value.type());
+    std::string_view s;
+    switch (type) {
+      case simdjson::ondemand::json_type::string: {
+        SIMDJSON_ASSIGN_OR_RAISE(s, value.get_string());
+        break;
+      }
+      default:
+        return simdjson::INCORRECT_TYPE;
+    }
+    int32_t day = 0;
+    // If the value has fewer than four digits, it is interpreted as the number
+    // of days since January 1, 1970.
+    if (s.size() < 4) {
+      auto result = folly::tryTo<int32_t>(s);
+      if (!result.hasError()) {
+        day = result.value();
+      } else {
+        return simdjson::INCORRECT_TYPE;
+      }
+    } else {
+      auto days =
+          util::fromDateString(StringView(s), util::ParseMode::kSparkCast);
+      if (!days.hasError()) {
+        day = days.value();
+      } else {
+        if (stringImpl::stringPosition<true /*isAscii*/>(s, kGMT, 1) > 0) {
+          // Remove 'GMT'.
```

Comment:
Are several "GMTs" that are present at any position valid in Spark?  Is there any other allowed string?

Reply:
The Spark PR mentioned that all instances of GMT need to be removed, but the only example I found in the Spark repo is 2000-01-01T00:00GMT+01:00.
https://github.com/apache/spark/commit/979a8eb04ae1dec2928cc49028b3df03b4ba78ce
https://github.com/apache/spark/blob/418cfd1f78014698ac4baac21156341a11b771b3/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala#L72